### PR TITLE
Catalogue item propeties bug #476

### LINKED ISSUE #476: Catalogue categories allowed values bug
Animation

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

#476

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

closes #476

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
Fix the aria-label and improve testings

File: src/catalogue/category/catalogueCategoryDialog.component.tsx [hunk 1]
```diff
@@ -60,15 +62,54 @@ const CatalogueCategoryDialog = React.memo(
...
         is_leaf: false,
         catalogue_item_properties: undefined,
       });
 
     React.useEffect(() => {
-      if (selectedCatalogueCategory)
+      if (selectedCatalogueCategory) {
+        const updatedCatalogueItemProperties =
+          selectedCatalogueCategory.catalogue_item_properties?.map((item) => {
+            // Assuming allowed_values should be an array of objects with id and value keys
```
Comment (joshuadkitenge) [inline src/catalogue/category/catalogueCategoryDialog.component.tsx:76]:
```suggestion
            // transform allowed_values to an array of objects with id and value keys
```

File: src/catalogue/category/cataloguePropertiesForm.component.tsx [hunk 2]
```diff
@@ -58,68 +62,92 @@ function CataloguePropertiesForm(props: CataloguePropertiesFormProps) {
...
+    ];
+
+    // Find the index of the item with the given cip_placement_id
+    const index = updatedFormFields.findIndex(
+      (field) => field.cip_placement_id === cip_placement_id
+    );
+
+    if (index === -1) {
+      // Item with given cip_placement_id not found
+      return; // or handle the error accordingly
```
Comment (joshuadkitenge) [inline src/catalogue/category/cataloguePropertiesForm.component.tsx:83]:
```suggestion
      return;
```

File: src/catalogue/category/cataloguePropertiesForm.component.tsx [hunk 3]
```diff
@@ -58,68 +62,92 @@ function CataloguePropertiesForm(props: CataloguePropertiesFormProps) {
...
-    const updatedFormFields: CatalogueCategoryFormData[] = [...formFields];
+    const updatedFormFields: CatalogueCategoryFormDataWithPlacementIds[] = [
+      ...formFields,
+    ];
+
+    const fieldIndex = updatedFormFields.findIndex(
+      (field) => field.cip_placement_id === cip_placement_id
+    );
+    if (fieldIndex === -1) {
+      // Handle the case where the field with the given cip_placement_id is not found
```
Comment (joshuadkitenge) [inline src/catalogue/category/cataloguePropertiesForm.component.tsx:120]:
```suggestion
```

File: src/catalogue/category/cataloguePropertiesForm.component.tsx [hunk 4]
```diff
@@ -58,68 +62,92 @@ function CataloguePropertiesForm(props: CataloguePropertiesFormProps) {
...
+      ...formFields,
+    ];
+
+    // Find the index of the item with the given cip_placement_id
+    const index = updatedFormFields.findIndex(
+      (field) => field.cip_placement_id === cip_placement_id
+    );
+
+    if (index === -1) {
+      // Item with given cip_placement_id not found
```
Comment (joshuadkitenge) [inline src/catalogue/category/cataloguePropertiesForm.component.tsx:82]:
```suggestion
```

File: src/catalogue/category/cataloguePropertiesForm.component.tsx [hunk 5]
```diff
@@ -134,29 +162,48 @@ function CataloguePropertiesForm(props: CataloguePropertiesFormProps) {
...
+      return; // or handle the error accordingly
```
Comment (joshuadkitenge) [inline src/catalogue/category/cataloguePropertiesForm.component.tsx:193]:
```suggestion
      return;
```

File: src/catalogue/category/cataloguePropertiesForm.component.tsx [hunk 6]
```diff
@@ -178,84 +225,122 @@ function CataloguePropertiesForm(props: CataloguePropertiesFormProps) {
...
-      onChangeFormFields(updatedFormFields);
+    // Find the index of the field with the provided cip_placement_id
+    const fieldIndex = formFields.findIndex(
+      (field) => field.cip_placement_id === cip_placement_id
+    );
 
-      // Remove the error when the value is changed
+    if (fieldIndex === -1) {
+      // If the field with the provided cip_placement_id doesn't exist, return or handle the error
+      return; // Or handle the error accordingly
```
Comment (joshuadkitenge) [inline src/catalogue/category/cataloguePropertiesForm.component.tsx:239]:
```suggestion
    if (fieldIndex === -1) {
      return;
```

File: src/catalogue/category/cataloguePropertiesForm.component.tsx [hunk 7]
```diff
@@ -178,84 +225,122 @@ function CataloguePropertiesForm(props: CataloguePropertiesFormProps) {
...
+    av_placement_id: string
+  ) => {
+    // Find the index of the field with the provided cip_placement_id
+    const fieldIndex = formFields.findIndex(
+      (field) => field.cip_placement_id === cip_placement_id
+    );
+
+    if (fieldIndex === -1) {
+      // If the field with the provided cip_placement_id doesn't exist, return or handle the error
+      return; // Or handle the error accordingly
```
Comment (joshuadkitenge) [inline src/catalogue/category/cataloguePropertiesForm.component.tsx:303]:
```suggestion
    if (fieldIndex === -1) {
      return; 
```

**Review by joelvdavies — APPROVED** [review]
Works well, fixes the issue.

**Pull request merged**